### ISSUE TITLE
Model created with create_model do not execute each_item validator

### ISSUE THREAD

**Issue by Maksych** [pydantic/pydantic#4832]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Model created with create_model do not execute each_item validator

### Example Code

```Python
from typing import Any
import pydantic


def validate_item(v: Any) -> Any:
    print(v)
    return v


class A(pydantic.BaseModel):
    a: list[int]


validators = {
    "validate_a_1": pydantic.validator("a", allow_reuse=True)(validate_item)
}
B = pydantic.create_model("B", __base__=A, __validators__=validators)
B.parse_obj({"a": [1, 2, 3]})
# [1, 2, 3]

validators = {
    "validate_a_2": pydantic.validator(
        "a", each_item=True, allow_reuse=True
    )(validate_item),
}
B = pydantic.create_model("B", __base__=A, __validators__=validators)
B.parse_obj({"a": [1, 2, 3]})
# not output, validator not working

validators = {
    "validate_a_1": pydantic.validator("a", allow_reuse=True)(validate_item),
    "validate_a_2": pydantic.validator(
        "a", each_item=True, allow_reuse=True
    )(validate_item),
}
B = pydantic.create_model("B", __base__=A, __validators__=validators)
B.parse_obj({"a": [1, 2, 3]})
# not output, validators not working

```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.2
            pydantic compiled: True
                 install path: /home/ubuntu/projects/learn/py_learn/.venv/lib/python3.11/site-packages/pydantic
               python version: 3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:24:40) [GCC 10.4.0]
                     platform: Linux-5.15.79.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
     optional deps. installed: ['typing-extensions'
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [X] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [X] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [X] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by hramezani** [comment]
Thanks @Maksych for reporting the issue.

I can confirm that the validation doesn't work for the cases you've mentioned.
But, the validator will be executed if you create your model like the below:
```python
B = pydantic.create_model("B", a=(list, ...), __validators__=validators)
```

We are working on `V2` and I think it will not be fixed in `1.10.x` because the plant is to not work on `V1` and focusing on `V2`

**Comment by hramezani** [comment]
@Maksych as you may know, Pydantic `V2` is in the alpha state and your input is valuable for us.
Could you please test your case with V2? Please remember the old-style validators will be deprecated in Pydantic V2 and you need to change them. You can take a look at the [migration doc](https://github.com/pydantic/pydantic/blob/main/docs/migration.md).

I think your example code can be changed like:

```py
@validator('a', each_item=True)
def validate_item1(v: Any) -> Any:
    print(v)
    return v


class A(pydantic.BaseModel):
    a: list[int]


B = pydantic.create_model("B", __base__=A, __validators__={'validate_item1': validate_item1})
print(B.parse_obj({"a": [1, 2, 3]}))
```

Note: The `allow_reuse` is deprecated and does nothing.

You can use the new `validator` and `field_validator` decorators. see the [example code in migration doc](https://github.com/pydantic/pydantic/blob/main/docs/migration.md#changes-to-validators).
